### ISSUE TITLE
* aws_instance.instance: unexpected EOF

### ISSUE THREAD

**Issue by jbanmene** [terraform-providers/terraform-provider-aws#6532]
### Community Note

Hi there,

when i'm trying to launch a new instance,my terraform failed with error:
* aws_instance.Microsoft_Instance: unexpected EOF
A part of my config related to my instance  below 

### Terraform Version

Terraform v0.11.10
+ provider.aws v1.46.0
+ provider.template v1.0.0

### Affected Resource(s)

* aws_instances

### Terraform Configuration Files
my instance module

```
#Setting Provider
provider "aws" {
  region = "${var.region}"
}

#Setting User data
data "template_file" "user_data" {
  template = "${file("..\\..\\..\\Files\\user_data.txt")}"

  vars {
    envVariable     = "Vendor:${var.Vendor}#Product:${var.Product}#Action:${var.Action}#AvailableName:${var.Available_Name}#TargetOU:${var.Target_OU}"
    BootstrapPrefix = "${var.BootstrapPrefix}"
    Environment     = "${var.Environment}"
    SetupScripts    = "${var.SetupScripts}"
    ArchiveType     = "${var.ArchiveType}"
  }
}

#Setting aws Instance
data "aws_ami" "ami_base" {
filter {
  name = "tag:Name"
  values = ["${var.ami}"]
}
most_recent = true
}

resource "aws_instance" "Microsoft_Instance" {
  key_name             = "${var.key_name}"
  iam_instance_profile = "${var.iam_instance_profile}"
  ami                    = "${data.aws_ami.ami_base.id}"
  instance_type          = "${var.instance_type}"
  vpc_security_group_ids = ["${split(",",var.vpc_security_group_ids)}"]
  subnet_id              = "${var.subnet_id}"
  private_ip             = "${var.private_ip}"

  tags {
    Application = "${var.Tag_Application}"
    dependency  = "${var.Tag_Dependency}"
    depends_id  = "${var.depends_id}"
    Name = "${var.Available_Name}"
  }
  root_block_device {
    volume_type = "${var.volume_type}"
    volume_size = "${var.rootVolume_size}"
  }

  credit_specification {
    cpu_credits = "${var.cpu_credit}"
  }
  user_data = "${data.template_file.user_data.rendered}"
}
output "private_ip" {
  value = "${aws_instance.Microsoft_Instance.private_ip}"
}
```

### Panic Output

https://gist.github.com/jbanmene/642f5d8edecbd2abd6f50c92751aec6f

### Expected Behavior

create an instance

### Actual Behavior
Terraform crashes

### Steps to Reproduce
terraform apply

1. `terraform apply`

### References

* #0000

**Comment by radekg** [comment]
It could be related to your `template = "${file("..\..\..\Files\user_data.txt")}"`. I wonder if it would be fine with `template = "${file("${path.module}/../../../Files\user_data.txt")}"`

**Comment by jbanmene** [comment]
all my module calls the userdata that way, 
It is a windows machine 

I think I find the root cause, I change the provider version from 1.46. to 1.4.0 and my code works
maybe someone from terraform could confirm that aws provider 1.46 have issues

**Comment by jbanmene** [comment]
well it's not the provider version either, the crash is because of the  credit_specification , I launch a T3 instance and I didn't set the value for  cpu_credit.

**Comment by nywilken** [comment]
Hi @jbanmene are you still having issues here? There's been a number of changes to the `aws_instance` resource since this issue was reported. If you are still having issues would you mind testing with the latest provider release. 

If your issue has been resolved please close this issue; adding any relevant information as a comment.

Cheers!

**Comment by bflad** [comment]
This has been released in [version 2.16.0 of the Terraform AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#2160-june-20-2019). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
